viewing_party/party.py

```diff
@@ -1,23 +1,181 @@
+    sum_rating = 0
+    watched_movies = user_data["watched"]
+
+    if len(watched_movies) == 0: return 0.0
+
+    for movie in watched_movies:
+        sum_rating += movie["rating"]
+    average_rating = sum_rating / len(watched_movies)
+
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    highest_num = 0
+    highest_genre = ""
+    watched_movies = user_data["watched"]
+
+    if len(watched_movies) == 0: return None
+
+    for movie in watched_movies:
+        if movie["genre"] in genre_dict:
+            genre_dict[movie["genre"]] += 1
+        else:
+            genre_dict[movie["genre"]] = 1
+
+    for genre, num in genre_dict.items():
+        if num > highest_num:
+            highest_num = num
+            highest_genre = genre
+    
+    return highest_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_friend_movies(friend_list):
+    friend_watched_movies = []
+
+    for movies in friend_list:
+        for movie in movies["watched"]:
+            friend_watched_movies.append(movie)
+    
+    return friend_watched_movies
+
+def get_unique_watched(user_data):
+    watched_movies = user_data["watched"].copy()
+    friend_watched_movies = get_friend_movies(user_data["friends"])
+    not_watched_by_friends = []
+
+    fw_copy = friend_watched_movies.copy()
+
+    for movie in watched_movies:
+        if movie not in friend_watched_movies:
+            not_watched_by_friends.append(movie)
+
+    return not_watched_by_friends
+
+def get_friends_unique_watched(user_data):
+    watched_movies = user_data["watched"].copy()
+    friend_watched_movies = get_friend_movies(user_data["friends"])
+    not_watched_by_user = []
+    
+    for movies in user_data["friends"]:
+        for movie in movies["watched"]:
+            friend_watched_movies.append(movie)
+
+    fw_copy = friend_watched_movies.copy()
+
+    for movie in friend_watched_movies:
+        if movie not in watched_movies and movie not in not_watched_by_user:
+            not_watched_by_user.append(movie)   
+
+    return not_watched_by_user
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommended_movies = []
+    friends_unique_watched = get_friends_unique_watched(user_data)
+    friend_watch = get_friend_movies(user_data["friends"])
+
+    if len(user_data["watched"]) == 0 or len(friend_watch) == 0:
+        return []
+
+    for movie in friends_unique_watched:
+        if movie["host"] in user_data["subscriptions"]:
+            recommended_movies.append(movie)
+
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_user_genre(movie_list):

```

Comment:
Nice job making a helper function! But where do we already have this function? Let's use `get_most_watched_genre` rather than making another function that does the same thing.